iac-extensions/docker/src/main/java/org/sonar/iac/docker/parser/grammar/DockerGrammar.java

```diff
@@ -262,41 +275,113 @@ public UserTree USER() {
+   * Only one key supported as it match everything after the key as value.
+   */
+  public KeyValuePairTree KEYVALUEPAIR_FULL_LINE() {
+    return b.<KeyValuePairTree>nonterminal(DockerLexicalGrammar.KEYVALUEPAIR_FULL_LINE).is(
+      f.keyValuePair(
+        b.token(DockerLexicalGrammar.KEYVALUEPAIR_KEY),
+        b.token(DockerLexicalGrammar.KEYVALUEPAIR_VALUE_FULL_LINE)
+      )
+    );
+  }
+
+  // All KeyValuePairListTree
+  /**
+   * Match a single KeyValuePair with only the key defined, nothing else. Match it as a list
+   * Example : "key"
+   */
+  public KeyValuePairListTree KEYVALUEPAIRLIST_KEY_ONLY() {
+    return b.<KeyValuePairListTree>nonterminal(DockerLexicalGrammar.KEYVALUEPAIR_LIST_SINGLETON_KEY_ONLY).is(
+      f.keyValuePairSingletonList(KEY_ONLY())
+    );
+  }
+
+  /**
+   * Match a single KeyValuePair, considering the first word as the key and the rest as the value.
+   * Example : "key value1 value2 value3"
+   */
+  public KeyValuePairListTree KEYVALUEPAIRLIST_SINGLETON() {
+    return b.<KeyValuePairListTree>nonterminal(DockerLexicalGrammar.KEYVALUEPAIR_LIST_SINGLETON).is(
+      f.keyValuePairSingletonList(KEYVALUEPAIR_FULL_LINE())
     );
   }
 
   /**
-   * To match such element : key1 value1 value1bis value1tris
+   * Match a list of KeyValuePair separated by space(s)
+   * Example : "key1=value1 key2=value2"
    */
-  public KeyValuePairTree KEY_VALUE_PAIR() {
-    return b.<KeyValuePairTree>nonterminal(DockerLexicalGrammar.KEY_VALUE_PAIR_SINGLE).is(
-      f.keyValuePair(b.token(STRING_LITERAL), b.token(STRING_UNTIL_EOL))
+  public KeyValuePairListTree KEYVALUEPAIRLIST_SPACE_SEPARATOR() {
+    return b.<KeyValuePairListTree>nonterminal(DockerLexicalGrammar.KEYVALUEPAIR_LIST_SPACE).is(
+      f.keyValuePairList(
+        KEYVALUEPAIR_EQUALS(),
+        b.zeroOrMore(
+          f.tuple(
+            b.token(DockerLexicalGrammar.SPACE),
+            KEYVALUEPAIR_EQUALS()
+          )
+        )
+      )
     );
   }
 
   /**
-   * To match such element : key1=value1
+   * Match a list of KeyValuePair separated by comma. Possible KeyValuePair are "key only" and "key=value"
+   * Example : "key1=value1,key2,key3=value3"
    */
-  public KeyValuePairTree KEY_VALUE_PAIR_WITH_EQUALS() {
-    return b.<KeyValuePairTree>nonterminal(DockerLexicalGrammar.KEY_VALUE_PAIR_EQUALS).is(
-      f.keyValuePairEquals(b.token(KEY_IN_KEY_VALUE_PAIR_IN_EQUALS_SYNTAX), b.token(Punctuator.EQU), b.token(STRING_LITERAL))
+  public KeyValuePairListTree KEYVALUEPAIRLIST_COMMA_SEPARATOR() {
+    return b.<KeyValuePairListTree>nonterminal(DockerLexicalGrammar.KEYVALUEPAIR_LIST_COMMA).is(
+      f.keyValuePairList(
+        b.firstOf(
+          KEYVALUEPAIR_EQUALS(),
+          KEY_ONLY()
+        ),
+        b.zeroOrMore(
+          f.tuple(
+            b.token(DockerLexicalGrammar.COMMA),
+            b.firstOf(
+              KEYVALUEPAIR_EQUALS(),
+              KEY_ONLY()
+            )
+          )
+        )
+      )
```

Comment:
The case where we don't want any space before/inside this `KeyValuePairList`.
Typical use in ParamTree / RunTree : `RUN --mount=key=value`
No spaces are allowed between `mount=` and `key=value`